src/Util/ComposeFileManipulator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * Manipulate Docker Compose Files.
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ * @final
+ */
+class ComposeFileManipulator
+{
+    public const COMPOSE_FILE_VERSION = '3.7';
+
+    private $composeData;
+
+    public function __construct(string $contents = '')
+    {
+        if ('' === $contents) {
+            $this->composeData = $this->getBasicStructure();
+        } else {
+            $this->composeData = (new YamlSourceManipulator($contents))->getData();
+        }
+    }
+
+    public function getComposeData(): array
+    {
+        return $this->composeData;
+    }
+
+    public function serviceExists(string $name): bool
+    {
+        if (\array_key_exists('services', $this->composeData)) {
+            return \array_key_exists($name, $this->composeData['services']);
+        }
+
+        return false;
+    }
+
+    public function addDockerService(string $name, array $details): void
+    {
+        $this->composeData['services'][$name] = $details;
```

Comment:
Edge case: but in some insane universe, the `services` might not exist (like, if they created their own... and I guess had zero services to start)

Reply:
i always thought if `services` didn't exist, the above would create it w/o issue.. have i been living my life wrong this whole time?

Then again for arrays that works, but i can't remember what the yaml manipulator does in that situation.

Reply:
> i always thought if services didn't exist, the above would create it w/o issue.. have i been living my life wrong this whole time?

I just tried it, apparently *I* have been living my life wrong this whole time! 🙃